### PR TITLE
BUG: DataFrame[dt64].quantile(axis=1) when empty returning f8

### LINKED ISSUE #41544: BUG: Empty DataFrame(..., dtype='datetime64[ns]').quantile(..., axis=1) returns float64 dtype
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

This ideally should still return `datetime64[ns]` dtype

```
In [1]: df = DataFrame(columns=["a", "b"], dtype="datetime64[ns]")

In [2]: df
Out[2]:
Empty DataFrame
Columns: [a, b]
Index: []

In [3]: df.quantile(0.5, axis=1, numeric_only=False)
Out[3]: Series([], Name: 0.5, dtype: float64)

In [4]: df.quantile(0.5, axis=1, numeric_only=False).dtype
Out[4]: dtype('float64')

In [5]: df
Out[5]:
Empty DataFrame
Columns: [a, b]
Index: []

In [6]: df.T
Out[6]:
Empty DataFrame
Columns: []
Index: [a, b]

# Probably the reason why we might get float64 dtype afterwards
In [7]: df.T.dtypes
Out[7]: Series([], dtype: object)
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #41544
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -10460,27 +10461,41 @@ def quantile(
...
 
         if not is_list_like(q):
             # BlockManager.quantile expects listlike, so we wrap and unwrap here
-            res = self.quantile(
+            res_df = self.quantile(
                 [q], axis=axis, numeric_only=numeric_only, interpolation=interpolation
             )
-            return res.iloc[0]
+            res = res_df.iloc[0]
+            if axis == 1 and len(self) == 0:
```
Comment (jreback) [inline pandas/core/frame.py:10472]: can we do this in quantile itself?

**Review by jreback — CHANGES_REQUESTED** [review]
cc @mroeschke

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:10472]: no, these cases are only reached when we dont have any blocks to operate on.

**Review by mroeschke — APPROVED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (jreback) [reply pandas/core/frame.py:10474]: hmm really ought to push this down to eliminate this type of logic in the frame itself, ok for now.

**Review by jreback — APPROVED** [review]

**Pull request merged**